tqa_training_lib/trainers/tf_tweetqa_trainer.py

```diff
@@ -0,0 +1,48 @@
+import gc
+from transformers import TFBertForQuestionAnswering
+import tensorflow as tf
+
+from tqa_training_lib.trainers.tweetqa_training_args import TweetQATrainingArgs
+from tqa_training_lib.trainers.tweetqa_trainer import TweetQATrainer
+
+
+class TFTweetQATrainer(TweetQATrainer):
+    def train(self, train_encodings, val_encodings, args):
+        gpus = tf.config.list_physical_devices('GPU')
+        if gpus:
+            try:
+                # Currently, memory growth needs to be the same across GPUs
+                for gpu in gpus:
+                    tf.config.experimental.set_memory_growth(gpu, True)
+                    logical_gpus = tf.config.list_logical_devices('GPU')
+                    print(len(gpus), "Physical GPUs,", len(logical_gpus), "Logical GPUs")
+            except RuntimeError as e:
+                # Memory growth must be set before GPUs have been initialized
+                print(e)
+
+        train_dataset = tf.data.Dataset.from_tensor_slices((
```

Comment:
Does Tensorflow require two input dicts split as specified?  Any reason this isn't done as one dict?

Reply:
Good question, honestly I'm unsure of the specific requirements. Most of this was pulled from the interwebs. I remember also finding a github issue or huggingface forum discussion mentioning that start_logits and end_logits is required. I'll try to find both sources later today for more context.

Note that right now the val_dataset is unused. I'll add that to the todo list, it could be beneficial in training.